src/test/regress/expected/query_single_shard_table.out

```diff
@@ -1197,6 +1198,8 @@ SELECT COALESCE(raw_events_first.user_id, users_ref_table.user_id)
 FROM raw_events_first
      RIGHT JOIN (users_ref_table LEFT JOIN raw_events_second ON users_ref_table.user_id = raw_events_second.user_id)
      ON raw_events_first.user_id = users_ref_table.user_id;
+DEBUG:  cannot perform a lateral outer join when a distributed subquery references a reference table
```

Comment:
This is an odd message since we do not have lateral join. But it is fine after I check `DeferredErrorIfUnsupportedRecurringTuplesJoin`. Normally we expect only lateral recurring outer joins can throw error here since we recursively plan other recurring outer joins via recursive planner. But, this check can be hit by INSERT SELECT planner before we recursively plan SELECT part of the query.

Reply:
https://github.com/citusdata/citus/pull/6752#discussion_r1175404134

>  But it is fine after I check DeferredErrorIfUnsupportedRecurringTuplesJoin.

I think we should fix the wording used DeferredErrorIfUnsupportedRecurringTuplesJoin, i.e., we should stop assuming that the joins that appear in that function are lateral joins.